homeassistant/components/homeconnect/binary_sensor.py

```diff
@@ -0,0 +1,72 @@
+"""Provides a binary sensor for Home Connect.
+
+For more details about this platform, please refer to the documentation at
+https://home-assistant.io/integrations/binary_sensor.homeconnect/
+"""
+import logging
+
+from homeassistant.components.binary_sensor import BinarySensorDevice
+
+from .const import BSH_DOOR_STATE, DOMAIN
+from .entity import HomeConnectEntity
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the Home Connect binary sensor."""
+
+    def get_entities():
+        entities = []
+        hc_api = hass.data[DOMAIN][config_entry.entry_id]
+        for device_dict in hc_api.devices:
+            entity_dicts = device_dict.get("entities", {}).get("binary_sensor", [])
+            entity_list = [HomeConnectBinarySensor(**d) for d in entity_dicts]
+            device = device_dict["device"]
+            device.entities += entity_list
+            entities += entity_list
+        return entities
+
+    async_add_entities(await hass.async_add_executor_job(get_entities), True)
+
+
+class HomeConnectBinarySensor(HomeConnectEntity, BinarySensorDevice):
+    """Binary sensor for Home Connect."""
+
+    def __init__(self, device, name, device_class):
+        """Initialize the entity."""
+        super().__init__(device, name)
+        self._device_class = device_class
+        self._state = None
+
+    @property
+    def is_on(self):
+        """Return true if the binary sensor is on."""
+        return bool(self._state)
+
+    @property
+    def available(self):
+        """Return true if the binary sensor is available."""
+        return self._state is not None
+
+    def update(self):
```

Comment:
Does this method do I/O or could it be run in the event loop?

Reply:
Right, it doesn't do I/O. I can just to `async def async_update` without needing to `await` anything, right?

Reply:
Yes! :+1: